src/etc/inc/filter.inc

```diff
@@ -2313,7 +2313,9 @@ function filter_generate_user_rule_arr($rule) {
 	$line = filter_generate_user_rule($rule);
 	$ret['rule'] = $line;
 	$ret['interface'] = $rule['interface'];
-	if ($rule['descr'] != "" and $line != "") {
+	if ($rule['tracker'] != "" and $line != "") {
+		$ret['descr'] = "label \"" . fix_rule_label("USER_RULE: {$rule['tracker']}") . "\"";
```

Comment:
Yes, I'll extend the pfSense PHP module to export the firewall rules information, it will simplify the implementation of this feature.
Stay tuned...